### ISSUE TITLE
General discussion about AntiMicroX and further development.

### ISSUE THREAD

**Issue by pktiuk** [AntiMicroX/antimicrox#5]
Hey,

I think AntiMicroX is a very good project which doesn't get deserved attention.
A lot of work has been done to original AntiMirco, and it could currently without any problems replace original Antimicro which is still quite popular.
- [downloads from github](https://somsubhra.com/github-release-stats/?username=antimicro&repository=antimicro) approx 350k overall
- [downloads from SourceForge](https://sourceforge.net/projects/antimicro.mirror/files/stats/timeline) approx 18k this week
Compared with AntiMicroX [clones](https://somsubhra.com/github-release-stats/?username=AntiMicroX&repository=antimicrox) and [star history](https://star-history.t9t.io/?ref=producthunt#AntiMicroX/antimicroX)

But there are some problems with this app which are contributing to low popularity and bad accessibility.

In my opinion, they are:
1. It is difficult to find it - it would be good if AntiMicroX would be easier to find by someone who is looking for this type of software. It should be searchable at least on sites like [AlternativeTo](https://alternativeto.net/software/antimicrox/) or [SourceForge](sourceforge.net). Maybe we should also promote it a bit there (add something like "If you liked an app then like it there" and put it in README, or maybe in "About" window)
2. It is a very inaccessible - there is no ready to install package for Windows or macOS (it would be good to have also `rpm` package), most of the users have to compile it manually (it can be very difficult for people who are not so tech-savvy)
3. Its site is not very visually compelling - I think we should change README a bit (top part could contain some screens and installation guide) and we should change the icon, because it has low resolution and isn't very clean (as most of minimalistic and modern icons today). We could use something from [flaticon](www.flaticon.com), icons from there have a license that allows us to use them in our project (even if this project would be commercial).

Things (I think) should be done for now:
- [x] select new icon for an app
- [x] finish AntiMicroX's site on [AlternativeTo](https://alternativeto.net/software/antimicrox/) and (maybe) edit AntiMicro's page
- [x] add AntiMicroX to [SourceForge](https://sourceforge.net/projects/antimicrox//) 
- [ ] add installer for Windows: https://github.com/AntiMicroX/antimicroX/issues/4
- [ ] (optional) add portable version for Windows (it was implemented in old AntiMicro [link](https://github.com/AntiMicro/antimicro/blob/master/BuildOptions.md))
- [ ] add installer for macOS
- [x] change README (add simple installation instructions, and maybe some screenshots of an app)

I think some of these tasks should get their own issues.

What do you think about these tasks? Do you have any comments? Ideas? Something else should be done?

**To sum up:**
If you want to help you can:
- Like AntiMicroX's on [AlternativeTo](https://alternativeto.net/software/antimicrox/) and [SourceForge](https://sourceforge.net/projects/antimicrox/)
- Consider helping us with one of tasks mentioned here (currently Windows port is our top priority)

(This issue is a copy of original issue https://github.com/juliagoda/antimicroX/issues/130 from legacy AntiMicroX repo.)

**Comment by trymeouteh** [comment]
I think by getting a Windows installer would definitely help. I am a Linux user and I am glad this is made for Linux DEB installer, however most gamers use Windows.

And if there isn't already, I hope this is in the AUR for Arch based distros.

**Comment by trymeouteh** [comment]
And what can help is to Star this github project

**Comment by OlegPopow** [comment]
Forgive my impertinence, but I have to raise an aspect that is impossible to find information about here. You seem to be an undisputed champion of promotion and advertising. Cool, but... What about project development and fixing real problems? First of all, the question of future updates of the Qt. library. Probably you will have to make appropriate changes in the project to keep it working. There were also some unresolved bugs on the old repository. Are you going to do something about it or focus only on promotion?

**Comment by pktiuk** [comment]
@OlegPopow 
I understand your curiosity (But I think, the next time you should just open new issue, that's the point of tag: `question`).
Currently, I want to finish, what I have started, when old repository was still active.
I want to provide good packages to allow users avoiding compilation. 
I am also working on setting up and configuring this repository to make working with it easy (like for example: https://github.com/AntiMicroX/antimicroX/issues/10)
I am also going to make some cleanup in CMakeLists.txt

In terms of code development I want you to know I am not familiar with this code yet. There are a lot of classes in this project, almost none of them are documented at all.
For now, I have generated some docs using doxygen and I will start reading code to understand what is inside.
When I will briefly understand  how it works, I will start fixing bugs, which I will be able to reproduce, and later I am going to move on to other bugs and adding features.
In terms of preparations for future releases of QT I can only say: `I don't know` I know C++, but I am not very experienced with QT, so it may take some time before I will start thinking about it. 
Don't expect any rapid progress in this matter, because it is just my side project developed in free time next to full-time job and university projects.
I am thinking about keeping this project alive without losing traction gained by legacy AntiMicroX and the only help I have now is @gombosg.

**Comment by OlegPopow** [comment]
> In terms of code development I want you to know I am not familiar with this code yet. There are a lot of classes in this project, almost none of them are documented at all.

To be honest, I haven't met too many Open Source projects in which classes would be documented. And to be even more honest, I don't recall that such a thing would be practiced too often even in KDE projects - and a large group of people work there. The previous maintainer has been developing the program on his own for two years, and for lack of anyone to help, he focused on fixing the bug. It's hard to expect one person to take care of absolutely everything, right? Anyway, in big companies, it also happens that a newly recruited employee is thrown to continue the project after the previous one and then you also have to reckon with the fact that such documentation will be either residual... or there will be no such documentation at all.


> I am thinking about keeping this project alive without losing traction gained by legacy AntiMicroX and the only help I have now is @gombosg.

As above. I fully understand that your team is two people... but on the other hand it is twice as big as the previous one!

> Don't expect any rapid progress in this matter, because it is just my side project developed in free time next to full-time job and university projects.

Of course, no one expects a miracle. The foundation for the program to be usable and build correctly on different versions of Qt. Qt 6.0 is coming, so good luck remains!

**Comment by gombosg** [comment]
@OlegPopow 

> Anyway, in big companies, it also happens that a newly recruited employee is thrown to continue the project after the previous one and then you also have to reckon with the fact that such documentation will be either residual...

Yeah, it's called 'legacy project maintenance', and that's a large chunk of work at any fairly large company. :slightly_smiling_face: 

> Of course, no one expects a miracle. The foundation for the program to be usable and build correctly on different versions of Qt. Qt 6.0 is coming, so good luck remains!

Thanks for the suggestions. 

@sirlucjan  - after getting the previous repo archived

I simply don't understand all the finger pointing and blaming going around [here](https://github.com/juliagoda/antimicroX/commit/968c925bac6b9c06600376463a5e1bb2ffd6a696) in the previous repo, due to 1 (un)star.

I'd just use [Hanlon's razor](https://github.com/dwmkerr/hacker-laws#hanlons-razor) at this point: "Never attribute to malice that which is adequately explained by stupidity."

@juliagoda

The problem with archiving the repo without any contingency plan, links to forks is that existing users are now fairly cut off from further updates. They have to go through the hassle of finding actively maintained forks of a project that already went through several ones.

I believe that one core value of open-source is civilized collaboration, the fact that volunteers united together can create lasting value for users.

I appreciate all the useful work that you have done so far on the repo, and would like to do everything possible to keep a good relationship with you and previous antimicro maintainers, because this is what is the best for the users.

Kindly [revert this commit](https://github.com/juliagoda/antimicroX/commit/968c925bac6b9c06600376463a5e1bb2ffd6a696) or state some clear requirements towards us for listing forks in the README so that users know what options they have if they would like to keep using the software.

**Comment by OlegPopow** [comment]
@gombosg 

You seem to be a jolly good fellow so I will share some advice with you. In this case, personal email contact would look much better instead of calling previous contributors to the board. 

> I simply don't understand all the finger pointing and blaming going around 

Such matters are better handled privately. If you have any objections, just write an email and try to explain it outside the public space. 

It is different to point out and report a bug that was not here. By the way, users often report bugs at the address they find in the "Help" menu in the application. You've already made some changes, just tag along and the problem will largely resolve itself.

**Comment by gombosg** [comment]
@OlegPopow 

> Such matters are better handled privately.

It was a public discussion originally, so I intended to continue it publicly.

Also, Github doesn't disclose e-mail addresses, so email is not an option, at least not initially.
Feel free to send me your e-mail address to gombosg at disroot dot. org, then let's continue the conversation there.

**Comment by OlegPopow** [comment]
I will give you a little tip. It is enough that you apply such a small trick:

https://github.com/AntiMicroX/antimicroX/commit/01bdca9179feb990c28a500dde38bdcb54b6d806.patch

And there you go. You know where to write an email. No further conversation is necessary, I am just an outsider.,

May the Force be with you.

**Comment by RaTcHeT302** [comment]
i honestly don't see the point in using this version, i mean the original works perfectly fine, plus i'm not a fan of the theme change, i really like white themes, but if there's an option to use the original theme i guess that wouldn't be horrible, but personally the tool did the job it needed to

and the fact that this new version doesn't run on windows is a big deal breaker, since i can't even see if it's any better than the original, plus i'm not a fan of this wierd drama that's going on and it's sorta off putting

this doesn't scream "long term project" to me and i feel like depending on something like this is risky since, i don't know, u guys must just wake up one day and u might drop it, but with the old version, i know that it'll always be there, and i don't need to worry about it not working in the future, this new project on the other hand

 i dunno, it just feels like a gamble to me, i just need a realiable tool to solve the job, i'm not sure what other improvements could be made to it, and i'm not close minded but, the whole thing is just weird as a whole

**Comment by RaTcHeT302** [comment]
bear in mind that i would be more open minded if i could run this on windows and actually see the changes for myself, but this sorta reminds me of that PS3 controller project, where the whole thing got so bloated, to the point where the original author just abbandoned the project and left everyone with a messier tool than the original, even though the original PS3 controller tool does its job just as well (while being easier to use, ironically)

**Comment by RaTcHeT302** [comment]
also i'm not a fan of how spread out this is, supposedly there are 3 repositories, and they all look very samey, this to me just screams messy as hell

sorry guys, i appreciate the thought but i'll stick to the original, i feel like ya'll are just clashing with each other, that's the feeling i'm getting anyway, this feels like a competition more to me, instead of, ppl trying to improve a really useful generalist tool

**Comment by pktiuk** [comment]
@RaTcHeT302 
> i'm not a fan of the theme change

Theme is the same as always, visible theme depends on your system configuration, most of qt-based apps can be switched to theme matching system-wide one without a hassle. I was doing these screenshots on my Ubuntu 20.04 with dark system-wide theme enabled.

> new version doesn't run on windows is a big deal breaker

I assume there, you are comparing this project with [legacy antimicro](https://github.com/AntiMicro/antimicro), but not with [previous version](https://github.com/juliagoda/antimicroX) of this fork.   
All I can tell, I fully agree that dropping Windows support by previous maintainer was a terrible decision, and we are going to restore it, this is one of our top priorities (but unluckily WIndows is not as developer-friendly as Linux :disappointed: and it may take some time).

> this doesn't scream "long term project" to me and i feel like depending on something like this is risky since, i don't know

I understand your point of view
This particular repository is just a fresh fork of an [abandoned fork ](https://github.com/juliagoda/antimicroX)of an [abandoned fork](https://github.com/AntiMicro/antimicro) :wink:  
At this stage of this project we are just cleaning up the repository and preparing it for further work.

>  i'm not sure what other improvements could be made to it, and i'm not close minded but, the whole thing is just weird as a whole

To check currently added features and fixes you can look into [changelog](https://github.com/AntiMicroX/antimicrox/blob/master/Changelog) or release sites [1](https://github.com/juliagoda/antimicroX/releases) [2](https://github.com/antimicrox/antimicroX/releases), most of these changes were just bugfixes.

> sorry guys, i appreciate the thought but i'll stick to the original, i feel like ya'll are just clashing with each other

If you are a Windows user you even can't switch to antimicrox yet :(  
This project has a bit of difficult history in terms of abandonment and changing maintainers (the same thing happened when antimicrox was forked from antimicro), but there are no competing teams developing the same piece of software in two places (this project is developed only here and this is the only place where you can get some support in case of any problems). The only issue was changing mainainer after creating fork of abandoned project (I partially understand it, because it may be difficult to let someone else continue project you put multiple hours into).

**Comment by OlegPopow** [comment]
> i honestly don't see the point in using this version, i mean the original works perfectly fine, plus i'm not a fan of the theme change, i really like white themes, but if there's an option to use the original theme i guess that wouldn't be horrible, but personally the tool did the job it needed to

You will no longer compile the old version on any current Linux distribution. The QT library has changed in such a way that the version you are so praising on Linux is useless.

> and the fact that this new version doesn't run on windows is a big deal breaker, since i can't even see if it's any better than the original, plus i'm not a fan of this wierd drama that's going on and it's sorta off putting

The author of AntimicroX (I mean the previous, not current) does not have and does not use Windows. She has marked it many times.  

If this version works on Linux and the previous one on Windows, what is the problem? You have a lot of alternatives on Windows and on Linux it's basically the only tool worth your attention.  Completely needless butthurt.

**Comment by OlegPopow** [comment]
> this doesn't scream "long term project" to me and i feel like depending on something like this is risky since, i don't know, u guys must just wake up one day and u might drop it, but with the old version, i know that it'll always be there, and i don't need to worry about it not working in the future, this new project on the other hand

Yes, the old version has not been developed for several years. You may be surprised, but previous developers of the original project left more than 200 bugs open and abandoned the project without any information. You don't want to tell anyone that the previous one was so perfect that it didn't require any changes and could be left as this perfect version. 

As I mentioned before, do not compare an apple with a pear. You won't use the old version on Linux anymore and a big thank you to the author for doing something about it. It's easier to talk to the pharmacists and to pick on anything.

**Comment by OlegPopow** [comment]
> All I can tell, I fully agree that dropping Windows support by previous maintainer was a terrible decision

You know, you could always pack the old AntiMicro and change the README.md file there...

**Comment by pktiuk** [comment]
@OlegPopow 
> You know, you could always fork the old AntiMicro and change the README.md file there

Yes, I know it, but you should also know dropping Windows support is not the only difference between antimicrox and antimicro, previous maintainers fixed tons of bugs, implemented support for Qt5 and added a lot of cool features.

> On the one hand you say that abandoning Windows support was a terrible decision and on the other hand you think that you need support from a developer who uses Windows on a daily basis.

Where do you see contradictions? I don't use Windows as a developer (but I own a license for one), but I am aware of huge Windows user base.   
I won't tell, that dropping support for 95% of potential users was a good decision.

**Comment by OlegPopow** [comment]
> Yes, I know it, but you should also know dropping Windows support is not the only difference between antimicrox and antimicro, previous maintainers fixed tons of bugs, implemented support for Qt5 and added a lot of cool features.
> 
> >

MAINTAINER, not maintainers. 

> Where do you see contradictions? I don't use Windows as a developer (but I own a license for one), but I am aware of huge Windows user base.

It's quite simple. If you don't use it every day, you probably understand that the previous one didn't use it either and therefore was not able to provide support. When you were asked for help, you mentioned it:

> I have copied missing files and CMakeLists configs from antimicro, but currently I have some problems with finding QT libraries by cmake.
In case if someone more familiarized with Windows would like to try this porting I have published my changes here:

If you were not able to do it yourself as a Linux user and you needed the help of someone from the outside, why is it so surprising?

As I wrote - MAINTAINER, not maintainers. One person can't take over everything. I am impressed that she managed to run such a project on her own for so long. 

With all due respect, you appeared at the very end of the previous project and your changes were so illusory that writing "MAINTAINERS" would be an abomination. Porting to QT5 and new features is 99.99% thanks to one person

To end this sterile discussion - it is probably obvious that without a computer with Windows installed, it is not possible to implement new features from the Linux version on this Windows. 
One person can't grasp everything and I have no idea why the hell it's so hard to understand.

**Comment by pktiuk** [comment]
I don't deny her accomplishments, but I don't find removing tons of files/configs/lines of code used by windows was necessary to continue development.
Migration was made between QT4 and QT5, both of them are supported by multiple systems, and properly written `CMakeLists.txt` can easily provide support for them systems (using only Linux I have written some apps which were later supported by Windows). 
Continuing this discussion doesn't make any sense.

**Comment by OlegPopow** [comment]
> and properly written `CMakeLists.txt` can easily provide support for them systems

So you have the opportunity to write it. Good luck, because you will need it.

**Comment by pktiuk** [comment]
Content of this issue doesn't match its title anymore, I will soon create new ones for discussions (with clearly defined rules and topics to avoid spam offtopic messages and sterile discussions).

**Comment by gombosg** [comment]
IDK why this was closed and #35 opened with the same title, but we all can continue the discussion there.

**Comment by pktiuk** [comment]
There was a lot of spam and mixed topics, there were some about original subject, abd some from offtop, changing description would rip fist part of discussion of context. If you think this mess would be a good start point for further discussion, then reopen this issue and close the new one (or you can work a bit as a moderator and remove things you consider as a spam and reopen it).
